controllers/aggregated_clusteroperator_controller.go

```diff
@@ -0,0 +1,148 @@
+	coWriter := aggregatedco.NewWriter(a.Configv1Client)
+
+	aggregatedCO := &openshiftconfigv1.ClusterOperator{}
+	if err := a.Get(ctx, req.NamespacedName, aggregatedCO); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	defer func() {
+		if err := coWriter.UpdateStatus(aggregatedCO, coBuilder.GetStatus()); err != nil {
+			log.Error(err, "error updating CO status")
+		}
+	}()
+
+	// Set the default CO status conditions: Progressing True, Degraded False, Available False
+	coBuilder.WithProgressing(openshiftconfigv1.ConditionTrue, "")
+	coBuilder.WithDegraded(openshiftconfigv1.ConditionFalse)
+	coBuilder.WithAvailable(openshiftconfigv1.ConditionFalse, "", "")
+
+	poList := &platformv1alpha1.PlatformOperatorList{}
+	if err := a.List(ctx, poList); err != nil {
+		log.Error(err, "error listing platformoperators")
+		return ctrl.Result{}, err
+	}
+
+	if len(poList.Items) == 0 {
+		// No POs on cluster, everything is fine
+		coBuilder.WithAvailable(openshiftconfigv1.ConditionTrue, "", "No POs Found")
+		return ctrl.Result{}, nil
+	}
+
+	statusErrorCheck := a.inspectPlatformOperators(poList)
+	if statusErrorCheck != nil {
+		// One of the POs is in an error state
+		// Update the Aggregated CO with the information on the failed PO
+		coBuilder.WithDegraded(openshiftconfigv1.ConditionTrue)
+		coBuilder.WithAvailable(openshiftconfigv1.ConditionFalse, utilerror.NewAggregate(statusErrorCheck.FailingErrors).Error(), "PO In An Error State")
+		return ctrl.Result{}, nil
+	}
+
+	coBuilder.WithAvailable(openshiftconfigv1.ConditionTrue, "All POs in a successful state", "POs Are Healthy")
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (a *AggregatedClusterOperatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&openshiftconfigv1.ClusterOperator{}, builder.WithPredicates(predicate.NewPredicateFuncs(func(object client.Object) bool {
+			return object.GetName() == aggregateCOName
+		}))).
+		Watches(&source.Kind{Type: &platformv1alpha1.PlatformOperator{}}, handler.EnqueueRequestsFromMapFunc(util.RequeueBundleDeployment(mgr.GetClient()))).
+		Complete(a)
+}
+
+type POStatusErrors struct {
+	FailingPOs    []*platformv1alpha1.PlatformOperator
+	FailingErrors []error
+}
+
+// inspectPlatformOperators iterates over all the POs on the cluster
+// and determines whether a PO is in a failing state by inspecting its status.
+// A nil return value indicates no errors were found with the POs provided.
+func (a *AggregatedClusterOperatorReconciler) inspectPlatformOperators(POList *platformv1alpha1.PlatformOperatorList) *POStatusErrors {
+	POstatuses := new(POStatusErrors)
+
+	for _, po := range POList.Items {
+		po := po.DeepCopy()
+		status := po.Status
+
+		for _, condition := range status.Conditions {
+			if condition.Reason == platformv1alpha1.ReasonSourceFailed || condition.Reason == platformv1alpha1.ReasonApplyFailed {
```

Comment:
nit: I think we'll end up consolidating this into a single status condition. I'd update this to just care about the ReasonApplyFailed for now. Alternatively, we could just iterate over the list of POs returned from the list query, and check whether `meta.FindStatusCondition(po.Status.Conditions, platformv1alpha1.TypeApplied`) is reporting a non metav1.ConditionTrue status.